kwave/utils/mapgen.py

```diff
@@ -693,8 +738,14 @@ def make_disc(grid_size: Vector, center: Vector, radius, plot_disc=False):
     return disc
 
 
-def make_circle(grid_size: Vector, center: Vector, radius: int, arc_angle: Optional[float] = None,
-                plot_circle: bool = False) -> np.ndarray:
+@beartype
+def make_circle(
+        grid_size: Vector, 
+        center: Vector, 
+        radius: Union[int, Int, Float], 
+        arc_angle: Optional[float] = None,
+        plot_circle: bool = False
+) -> NDArray[Shape["Dim1, Dim2"], Int]:
```

Comment:
Circle is defined as `circle = np.zeros(grid_size, dtype=int)`. This can be converted to `bool` but I'd like to keep function refactor outside of the PR scope.

Reply:
I changed the response type to use predefined type though.